### ISSUE TITLE
Chained collection and underlying db.object become unsynced.

### ISSUE THREAD

**Issue by lcaballero** [typicode/lowdb#32]
I'm not sure if this is intended.  What is happening though is that the db.object doesn't match the chained version of the same object over time as altering methods are applied to the chain because each new chain produces only a copy of the wrapper, and though this wrapper can be 'realized' into the right form, that value never gets assigned to db.object.

Here is a 'failing' test showing what I mean:

```
    describe('Autosave syncing with db.object', function() {

      it("both db.object and db('coll').value() should be equal after alteration to underlying data", function() {
        var db = low(file3, { async: false });
        var c1 = db('stay-synced');
        c1.push({id: 1});
        c1.push({id: 2});

        assert(c1.size().value() === 2);

        c2 = c1.remove({})

        assert(c2.size().value() === 0);

        assert.equal(c2.size().value(), db.object['stay-synced'].length);
      })
    });


```

I think that a better approach might be to add default mixins that includes a 'save'.  This would get around another problem: Every single method is composed with the save so even a find, or a pluck or a where, etc call the 'save()' method.  Additionally, it might not be a good idea to expose db.object, at least not directly because syncing back to the chain seems like it might be pretty complex.  I've got a local version that syncs db.object with the chain value after every chained method, but that doesn't seem appropriate either (because of find, pluck, filter, etc.)

**Comment by typicode** [comment]
Hi Lucas,

First sorry for the late reply. 

If I understand correctly, you'd like `remove` to modify database state and not simply create a copy?
Actually, it's "somehow" intended, it's the way Lo-Dash work, but I agree maybe not very convenient. `remove` always creates a copy of the array.

Maybe in https://github.com/TheNodeILs/lodash-contrib there would be functions that can change directly the array.

If you don't mind having ids in your database, https://github.com/typicode/underscore-db removes directly elements from the array.

About `.save()`, being called too often, I agree it's not optimal. I was hoping for Node 0.12 to be released soon to be able to use https://github.com/typicode/jsop. But since there's no sign of it, I'll add a list of "do not save" functions instead (typically find, pluck, filter, ...).

I'm currently working on LowDB v0.7 here https://github.com/typicode/lowdb/tree/0.7.0 which will have Lo-Dash 3.0.0 support and lazy-evaluation (less calls to `save()` too).

I'll think about a method for reassigning/resyncing arrays.

**Comment by typicode** [comment]
Just as an update, I've released `v0.7.0` and updated to the latest version of lodash.
It now does a quick checksum to see if database should be written to disk.
Also, when chaining, it now lazy evaluates, so you get only call to save() even though you chain many methods.
